fortify-issue-manager/src/main/java/com/microfocus/security/automation/fortify/issue/tracker/OctaneTracker.java

```diff
@@ -0,0 +1,125 @@
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public final class OctaneTracker implements BugTracker {
+    private final static JsonParser parser = new JsonParser();
+    private final String apiUrl;
+    private final String browseUrl;
+    private final String defectUrl;
+    private final OctaneTrackerClient client;
+
+    public OctaneTracker(final OctaneBugTrackerSettings bugTrackerSettings) {
+        apiUrl = bugTrackerSettings.getApiUrl();
+        browseUrl = String.format(
+            "/ui/entity-navigation?p=%s/%s&entityType=work_item&id=",
+            bugTrackerSettings.getSharedSpaceId(),
+            bugTrackerSettings.getWorkspaceId()
+        );
+        defectUrl = String.format(
+            "api/shared_spaces/%s/workspaces/%s/defects",
+            bugTrackerSettings.getSharedSpaceId(),
+            bugTrackerSettings.getWorkspaceId()
+        );
+        client = new OctaneTrackerClient(bugTrackerSettings);
+    }
+
+    @Override
+    public String createBug(final String payload) throws BugTrackerException {
+        try {
+            final String issue = client.performPostRequest(defectUrl, payload);
+
+            // Parse the Response
+            final JsonObject response = parser.parse(issue).getAsJsonObject();
+            if (response.has("data") && response.get("data").isJsonArray()) {
+                final JsonArray data = response.getAsJsonArray("data");
+                if (data.size() == 0) {
+                    throw new BugTrackerException("Issue was not created from payload: " + payload);
+                }
+                final JsonObject object = data.get(0).getAsJsonObject();
+                final String bugLink = object.get("id").getAsString();
+                return apiUrl + browseUrl + UrlEscapers.urlPathSegmentEscaper().escape(bugLink);
+            } else {
+                final String errors = response.get("errors").toString();
+                throw new BugTrackerException(errors);
+            }
+        } catch (final IOException e) {
+            throw new BugTrackerException(e);
+        }
+    }
+
+    @Override
+    public String getIssueDescription(final String fortifyIssueUrl, final List<Vulnerability> vulnerabilities) {
+        Collections.sort(vulnerabilities,
+            Comparator.comparing(Vulnerability::getPrimaryLocation).thenComparing(Vulnerability::getId));
+        final StringBuilder issues = new StringBuilder();
+        issues.append("<table><body><tr><th>&nbsp;Issue Id&nbsp;</th><th>&nbsp;Description&nbsp;</th></tr>");
+        for (final Vulnerability vulnerability : vulnerabilities) {
+            issues.append("<tr>")
+                .append("<td>&nbsp;<a href=\""
+                    + fortifyIssueUrl
+                    + vulnerability.getId()
+                    + "\">" + vulnerability.getId()
+                    + "</a>&nbsp;</td>")
+                .append("<td>&nbsp;" + vulnerability.getPrimaryLocation());
```

Comment:
Mixing string concatenation in the middle of `StringBuilder.append()`s is not as performant as just sticking with the `.append()` calls (because internally the string concatenation will start a second string builder).
```suggestion
            issues.append("<tr><td>&nbsp;<a href=\"")
                .append(fortifyIssueUrl)
                .append(vulnerability.getId())
                .append("\">")
                .append(vulnerability.getId())
                .append("</a>&nbsp;</td><td>&nbsp;")
                .append(vulnerability.getPrimaryLocation());
```